insta/src/snapshot.rs

```diff
@@ -419,7 +495,31 @@ impl Snapshot {
 
     /// Snapshot contents match another snapshot's.
     pub fn matches(&self, other: &Snapshot) -> bool {
-        self.contents() == other.contents()
+        match (self.contents(), other.contents()) {
+            (SnapshotContents::String(this), SnapshotContents::String(other)) => {
+                trim_content_str(this) == trim_content_str(other)
+            }
+            (
+                SnapshotContents::Binary {
+                    path: this,
+                    extension: this_ext,
+                },
+                SnapshotContents::Binary {
+                    path: other,
+                    extension: other_ext,
+                },
+            ) => {
+                if this_ext != other_ext {
+                    return false;
+                }
+
+                let mut this = File::open(this).unwrap();
+                let mut other = File::open(other).unwrap();
+
+                file_eq(&mut this, &mut other).unwrap()
```

Comment:
Historically we've had issues when running multiple tests at the same time; files can disappear under insta. Can we handle that a bit better here rather than blindly panicking? For example, at least with `.expect` with a message so we can debug?

Reply:
I've changed it to return `Result`'s with `Box<dyn Error>` to be more consistent with the rest of the code. It's not quite optimal for debugging because the origin of the error is missing, but I thought it would be better to be consistent.

Variants in a new error enum might be an alternative. Those could then also be returned as `dyn Error`'s. Or just using `expect` as you suggested. I'm not quite sure what is best here.

Reply:
Yes agree there are tradeoffs, and just ignoring the errors might not be ideal. 

A more detailed approach would be to think through when errors are really errors. For example, if a pending snapshot goes missing, that's fine — another process might have removed it. In contrast, if a target snapshot isn't available that's probably unexpected.

But so far ignoring errors with reading files seems to have been OK in practice; possibly we could print a message to protect against confusing results (though maybe others will disagree, ofc welcome).